MmSupervisorPkg/Drivers/StandaloneMmIpl/PiSmmIpl.c

```diff
@@ -1754,9 +1766,16 @@ GetFullSmramRanges (
   ASSERT (*FullMmramRangeCount == TempMmramRangeCount);
   *FullMmramRangeCount += AdditionMmramRangeCount;
 
-  FreePool (SmramRanges);
-  FreePool (SmramReservedRanges);
-  FreePool (TempSmramRanges);
+Exit:
+  if (SmramRanges != NULL) {
+    FreePool (SmramRanges);
+  }
+  if (SmramReservedRanges != NULL) {
+    FreePool (SmramReservedRanges);
+  }
+  if (TempSmramRanges) {

```

Comment:
Any reason why this check is different from the other two?

Reply:
Because this is a bug :). Nice catch.

Reply:
Fixed. Thanks again!